# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/DOMMediaStream.cpp
# Commit: 3f2015f34fc2
# Full Hash: 3f2015f34fc2daaaff9d90b7114157face56edf2
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2022-09-14 04:09:22
# Description:
#   Bug 1713940, null check owner before using it, r=padenot
#   
#   Reused an existing wpt. Added also some .step() calls there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D157143
# ==============================================================================

diff -r dff049971660 -r 3f2015f34fc2 dom/media/DOMMediaStream.cpp
--- a/dom/media/DOMMediaStream.cpp	Tue Sep 13 17:08:46 2022 +0000
+++ b/dom/media/DOMMediaStream.cpp	Tue Sep 13 17:09:24 2022 +0000
@@ -402,6 +402,9 @@
 }
 
 already_AddRefed<nsIPrincipal> DOMMediaStream::GetPrincipal() {
+  if (!GetOwner()) {
+    return nullptr;
+  }
   nsCOMPtr<nsIPrincipal> principal =
       nsGlobalWindowInner::Cast(GetOwner())->GetPrincipal();
   for (const auto& t : mTracks) {